legacy/firmware/ethereum.c

```diff
@@ -37,14 +37,14 @@
 #include "transaction.h"
 #include "util.h"
 
-/* maximum supported chain id.  v must fit in an uint32_t. */
-#define MAX_CHAIN_ID 2147483629
+/* maximum supported chain id.  v must fit in an uint64_t. */
+#define MAX_CHAIN_ID 9223372036854775807
```

Comment:
I think this should be (2^64 - 36) / 2:
```suggestion
#define MAX_CHAIN_ID 9223372036854775789
```

Reply:
Fixed :)

Reply:
Was off by 1 - updated to (2^64 - 35) / 2